coralogix/dashboard_widgets/hexagon.go

```diff
@@ -0,0 +1,813 @@
+			},
+			"custom_unit": schema.StringAttribute{
+				Optional:            true,
+				MarkdownDescription: "A custom unit",
+			},
+			"data_mode_type": schema.StringAttribute{
+				Optional: true,
+				Computed: true,
+				Validators: []validator.String{
+					stringvalidator.OneOf(DashboardValidDataModeTypes...),
+				},
+				Default: stringdefault.StaticString("unspecified"),
+			},
+			"thresholds": schema.SetNestedAttribute{
+				Optional: true,
+				Computed: true,
+				NestedObject: schema.NestedAttributeObject{
+					Attributes: map[string]schema.Attribute{
+						"from": schema.NumberAttribute{
+							Required: true,
+						},
+						"color": schema.StringAttribute{
+							Optional: true,
+						},
+						"label": schema.StringAttribute{
+							Optional: true,
+						},
+					},
+				},
+			},
+			"threshold_type": schema.StringAttribute{
+				Optional: true,
+				Computed: true,
+				Validators: []validator.String{
+					stringvalidator.OneOf(DashboardValidThresholdTypes...),
+				},
+				Default:             stringdefault.StaticString("unspecified"),
+				MarkdownDescription: fmt.Sprintf("The threshold type. Valid values are: %s.", strings.Join(DashboardValidThresholdTypes, ", ")),
+			},
+			"time_frame": TimeFrameSchema(),
+			"query": schema.SingleNestedAttribute{
+				Required: true,
+				Attributes: map[string]schema.Attribute{
+					"logs": schema.SingleNestedAttribute{
+						Attributes: map[string]schema.Attribute{
+							"lucene_query": schema.StringAttribute{
+								Optional: true,
+							},
+							"group_by": schema.ListNestedAttribute{
+								NestedObject: schema.NestedAttributeObject{
+									Attributes: ObservationFieldSchema(),
+								},
+								Optional: true,
+							},
+							"filters":     LogsFiltersSchema(),
+							"aggregation": LogsAggregationSchema(),
+						},
+						Optional: true,
+						Validators: []validator.Object{
+							objectvalidator.ExactlyOneOf(
+								path.MatchRelative().AtParent().AtName("metrics"),
+								path.MatchRelative().AtParent().AtName("spans"),
+								path.MatchRelative().AtParent().AtName("data_prime"),
+							),
+						},
+					},
+					"metrics": schema.SingleNestedAttribute{
+						Attributes: map[string]schema.Attribute{
+							"promql_query": schema.StringAttribute{
+								Required: true,
+							},
+							"promql_query_type": schema.StringAttribute{
+								Optional: true,
+								Computed: true,
+								Default:  stringdefault.StaticString(UNSPECIFIED),
+							},
+							"filters": MetricFiltersSchema(),
+							"aggregation": schema.StringAttribute{
+								Optional: true,
+								Computed: true,
+								Default:  stringdefault.StaticString("unspecified"),
+								Validators: []validator.String{
+									stringvalidator.OneOf(DashboardValidHexagonMetricAggregations...),
+								},
+							},
+						},
+						Optional: true,
+						Validators: []validator.Object{
```

Comment:
You don't have  to repeat this validation (objectvalidator.ExactlyOneOf) for every query type. It's enough to add it only for the first one.
This is a mistake I done in the past and it just adds unnecessary code...